### PR TITLE
Class#3 Homework

### PR DESCRIPTION
**Pull request by fgr-17**
* Create your own West application
* Must depend on Zephyr project
* only include cmsis and hal_nordic
* Use T2 topology
* CPP17 standart
* check if BOARD corresponds to host board and print a message to console

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 
-target_sources(app PRIVATE src/main.c)
\ No newline at end of file
+target_sources(app PRIVATE src/main.cpp)
+
+set(ALLOWED_BOARDS "nrf5340dk_nrf5340_cpuapp;native_posix_64")
+list(FIND ALLOWED_BOARDS ${BOARD} BOARD_INDEX)
+if(BOARD_INDEX EQUAL -1)
+  message(FATAL_ERROR 
+    "This application is intended to be built only for the following boards: ${ALLOWED_BOARDS}. Current BOARD is: ${BOARD}")
```
Comment (avashchylau-io) [inline CMakeLists.txt:15]: @fgr-17 Very nice! But I simply meant comparing CONFIG_BAORD definition in the source file (main.cpp) with the one of these strings. My fault :(

Comment (fgr-17) [reply CMakeLists.txt:15]: thanks! will review in the following PR

File: west.yml [hunk 2]
```diff
@@ -2,14 +2,25 @@ manifest:
...
+    remote: zephyrproject
 
   projects:
     # Zephyr repository (specific tag or branch)
     - name: zephyr
       path: zephyr
-      remote: zephyrproject
       revision: v3.6.0
       import: true  # Pull in Zephyr modules automatically
+    - name: cmsis
```
Comment (avashchylau-io) [inline west.yml:14]: Besides that, the current implementation of the west.yml will still clone all other dependencies that are not intended to be used. Go through the available properties that are used for _import_ to get only those 2 modules.

Explicit definition of the module will override the parent's definition (path/revision etc.). The other dependencies will still be cloned.

Comment (avashchylau-io) [reply west.yml:14]: The rest of the code LGTM!

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,9 +1,16 @@
 cmake_minimum_required(VERSION 3.20.0)
 
 set(APP "zephyr_app_scratch")
-
+set(CXX_STANDARD C++17)
```
Comment (avashchylau-io) [inline CMakeLists.txt:4]: You may also consider `CONFIG_STD_CPP17` option

**Pull request merged**